src/Frame.jsx

```diff
@@ -0,0 +1,142 @@
+import React, { PropTypes } from 'react';
+import ReactDOM from 'react-dom';
+import DocumentContext from './DocumentContext';
+
+const hasConsole = typeof window !== 'undefined' && window.console;
+const noop = () => {};
+let swallowInvalidHeadWarning = noop;
+let resetWarnings = noop;
+
+if (hasConsole) {
+  const originalError = console.error; // eslint-disable-line no-console
+  // Rendering a <head> into a body is technically invalid although it
+  // works. We swallow React's validateDOMNesting warning if that is the
+  // message to avoid confusion
+  swallowInvalidHeadWarning = () => {
+    console.error = (msg) => {  // eslint-disable-line no-console
+      if (/<head>/.test(msg)) return;
+      originalError.call(console, msg);
+    };
+  };
+  resetWarnings = () => (console.error = originalError);  // eslint-disable-line no-console
+}
+
+export default class Frame extends React.Component {
+  // React warns when you render directly into the body since browser extensions
+  // also inject into the body and can mess up React. For this reason
+  // initialContent initialContent is expected to have a div inside of the body
+  // element that we render react into.
+  static propTypes = {
+    style: PropTypes.object, // eslint-disable-line
+    head: PropTypes.node,
+    initialContent: PropTypes.string,
+    mountTarget: PropTypes.string,
+    contentDidMount: PropTypes.func,
+    contentDidUpdate: PropTypes.func,
+    children: PropTypes.oneOfType([
+      PropTypes.element,
+      PropTypes.arrayOf(PropTypes.element),
+    ]),
+  };
+
+  static defaultProps = {
+    style: {},
+    head: null,
+    children: null,
+    mountTarget: undefined,
+    contentDidMount: () => {},
+    contentDidUpdate: () => {},
+  };
+
+  static displayname = 'Frame';
```

Comment:
I wasn't aware of that. There might be a babel transform that I can include. I'll research this in the morning.